footer.php

```diff
@@ -185,17 +172,31 @@ function draw_navbuttons() {
         }
         if ($first) {
             list($icon, $bgcolour) = navbutton_get_icon($settings->buttonstype, 'first', $context, BLOCK_NAVBUTTONS_FIRSTICON, $settings->backgroundcolour, $settings->customusebackground);
-            $output .= make_navbutton($icon, $bgcolour, $first->name, $first->link);
+            $output .= make_navbutton($icon, $bgcolour, $first->name, $first->link, false, "first");
         }
     }
 
     if ($settings->prevbuttonshow && $prev) {
         list($icon, $bgcolour) = navbutton_get_icon($settings->buttonstype, 'prev', $context, BLOCK_NAVBUTTONS_PREVICON, $settings->backgroundcolour, $settings->customusebackground);
-        $output .= make_navbutton($icon, $bgcolour, get_string('prevactivity','block_navbuttons').': '.$prev->name, $prev->link);
+        $output .= make_navbutton($icon, $bgcolour, get_string('prevactivity','block_navbuttons').': '.$prev->name, $prev->link, false, "prec");
     }
+
+    if ($settings->homebuttonshow) {
+        $home = new stdClass;
+        if ($settings->homebuttontype == BLOCK_NAVBUTTONS_HOME_COURSE) {
+            $home->link = new moodle_url('/course/view.php', array('id'=>$COURSE->id));
+            $home->name = get_string('coursepage','block_navbuttons');
+        } else {
+            $home->link = $CFG->wwwroot;
+            $home->name = get_string('frontpage','block_navbuttons');
+        }
+        list($icon, $bgcolour) = navbutton_get_icon($settings->buttonstype, 'home', $context, BLOCK_NAVBUTTONS_HOMEICON, $settings->backgroundcolour, $settings->customusebackground);
+        $output .= make_navbutton($icon, $bgcolour, $home->name, $home->link, false, "home");
+    }
+
```

Comment:
I'm not convinced that moving the 'home' button down here is an improvement - I prefer it at the start.